src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -368,7 +375,28 @@ class Container extends React.Component<Props, ComponentState> {
                 return Util.formatMessageId(intl, FM.ENTITYCREATOREDITOR_FIELDERROR_DISTINCT)
             }
         }
-        return '';
+        return ''
+    }
+
+    onGetEnumErrorMessage = (value: string): string => {
+        const { intl } = this.props
+
+        if (value.length === 0) {
+            return ''
+        }
+
+        if (!/^[a-zA-Z0-9-]+$/.test(value)) {
+            return Util.formatMessageId(intl, FM.ENTITYCREATOREDITOR_FIELDERROR_ALPHANUMERIC)
+        }
+
+        return ''
+    } 
+
+    isEnumDuplicate = (value: string): boolean => {
+        if (!value) {
+            return false
+        }
+        return (this.state.enumValues.filter(v => v === value).length > 1)
```

Comment:
Minor, `some` would be better here instead of `filter`.  It implies we only care if 1 of the condition is true once.

Reply:
Talked offline, `some` doesn't work in this case because `enumValues` already has the new value in it. So we're trying to detect 2 of them due to addition of new value, not if it existed in the array before.